src/kikuchipy/signals/util/_master_pattern.py

```diff
@@ -99,86 +99,64 @@ def _get_direction_cosines_from_detector(
     """
     if detector.navigation_shape == (1,):
         pcx, pcy, pcz = detector.pc.squeeze().astype(np.float64)
+        gnomonic_bounds = detector.gnomonic_bounds.squeeze().astype(np.float64)
         func = _get_direction_cosines_for_fixed_pc
     else:
         pcx, pcy, pcz = detector.pc_flattened.T.astype(np.float64)
+        gnomonic_bounds = detector.gnomonic_bounds.reshape(
+            (np.prod(detector.navigation_shape), 4)
+        ).astype(np.float64)
         func = _get_direction_cosines_for_varying_pc
 
     if signal_mask is None:
         # Required since the Numba functions expect a boolean array
         signal_mask = np.ones(detector.size, dtype=bool)
 
     dc = func(
-        pcx=pcx,
-        pcy=pcy,
+        gnomonic_bounds=gnomonic_bounds,
         pcz=pcz,
         nrows=detector.nrows,
         ncols=detector.ncols,
-        tilt=detector.tilt,
-        azimuthal=detector.azimuthal,
-        sample_tilt=detector.sample_tilt,
+        u_s_inv=detector.u_s_inv,
         signal_mask=signal_mask,
     )
 
     return dc
 
 
 @njit(
-    "Tuple((float64, float64, float64, float64))(float64, float64, float64)",
-    cache=True,
-    nogil=True,
-    fastmath=True,
-)
-def _get_cosine_sine_of_alpha_and_azimuthal(
-    sample_tilt: float, tilt: float, azimuthal: float
-) -> tuple[float, float, float, float]:
-    alpha = (np.pi / 2) - np.deg2rad(sample_tilt) + np.deg2rad(tilt)
-    azimuthal = np.deg2rad(azimuthal)
-    return np.cos(alpha), np.sin(alpha), np.cos(azimuthal), np.sin(azimuthal)
-
-
-@njit(
-    (
-        "float64[:, :]"
-        "(float64, float64, float64, int64, int64, float64, float64, float64, bool_[:])"
-    ),
+    ("float64[:, :]" "(float64[:], float64, int64, int64, float64[:, ::1], bool_[:])"),
     cache=True,
     nogil=True,
     fastmath=True,
 )
 def _get_direction_cosines_for_fixed_pc(
-    pcx: float,
-    pcy: float,
+    gnomonic_bounds: np.ndarray,
     pcz: float,
     nrows: int,
     ncols: int,
-    tilt: float,
-    azimuthal: float,
-    sample_tilt: float,
+    u_s_inv: np.ndarray,
```

Comment:
Again, we should rename to something self-explanatory, such as `om_detector_to_sample` (`om` is used for orientation matrix in orix and elsewhere).